src/Stepping.Core/Stepping/Core/Steps/StepResolver.cs

```diff
@@ -5,42 +5,64 @@ namespace Stepping.Core.Steps;
 
 public class StepResolver : IStepResolver
 {
-    protected static Dictionary<string, Type>? CachedTypes { get; set; }
+    protected static Dictionary<string, StepResolverCachedTypeModel>? CachedTypes { get; set; }
 
     private static readonly object SyncObj = new();
 
     protected IServiceProvider ServiceProvider { get; }
+    protected IStepNameProvider StepNameProvider { get; }
 
-    public StepResolver(IServiceProvider serviceProvider)
+    public StepResolver(
+        IServiceProvider serviceProvider,
+        IStepNameProvider stepNameProvider)
     {
         ServiceProvider = serviceProvider;
+        StepNameProvider = stepNameProvider;
     }
 
-    public virtual async Task<IStep> ResolveAsync(string stepName) => (IStep)await InternalResolveAsync(stepName);
+    public virtual IStep Resolve(string stepName, object? args = null) => (IStep)InternalResolve(stepName, args);
 
-    protected virtual Task<object> InternalResolveAsync(string stepName)
+    public virtual IStep Resolve<TStep>(object? args = null) where TStep : IStep =>
+        (IStep)InternalResolve(StepNameProvider.Get<TStep>(), args);
+
+    protected virtual object InternalResolve(string stepName, object? args = null)
     {
-        if (CachedTypes is null)
-        {
-            lock (SyncObj)
-            {
-                CachedTypes ??= CreateCachedTypes();
-            }
-        }
+        TryWarmUp(StepNameProvider);
 
-        if (!CachedTypes.ContainsKey(stepName))
+        if (!CachedTypes!.ContainsKey(stepName))
         {
             throw new SteppingException("Invalid StepName.");
         }
 
-        return Task.FromResult(ServiceProvider.GetRequiredService(CachedTypes[stepName]));
+        return CachedTypes[stepName].HasArgs
+            ? ActivatorUtilities.CreateInstance(ServiceProvider, CachedTypes[stepName].Type, args!)
+            : ActivatorUtilities.CreateInstance(ServiceProvider, CachedTypes[stepName].Type);
+    }
+
+    protected static Dictionary<string, StepResolverCachedTypeModel> CreateCachedTypes(
+        IStepNameProvider stepNameProvider)
+    {
+        return AppDomain
```

Comment:
`AppDomain.CurrentDomain.GetAssemblies` returns only the currently loaded assemblies. It does not return all assemblies. Does it satisfy the purpose of this method?

https://docs.microsoft.com/en-us/dotnet/api/system.appdomain.getassemblies?view=net-6.0
https://stackoverflow.com/questions/10284861/not-all-assemblies-are-being-loaded-into-appdomain-from-the-bin-folder
https://stackoverflow.com/questions/2384592/is-there-a-way-to-force-all-referenced-assemblies-to-be-loaded-into-the-app-doma

Reply:
I have no idea. Do you know in which case will an assembly be not loaded but needed?

Reply:
> I have no idea. Do you know in which case will an assembly be not loaded but needed?

It's not relatively easy to describe.
There is a class library that is only used in a controller's action, and when the interface is not called, the assembly it is in is not loaded.

Is it possible to allow users to configure required assemblies?

Just like `AutoMapper`
https://github.com/AutoMapper/AutoMapper.Extensions.Microsoft.DependencyInjection/blob/master/src/AutoMapper.Extensions.Microsoft.DependencyInjection/ServiceCollectionExtensions.cs#L28

Reply:
> Is it possible to allow users to configure required assemblies?

Sometimes they forget to add new assemblies. I think registering steps in options is a better way, but it is still a burden for developers since they should always remember it.